forte/orbital-helpers/orbital_embedding.cc

```diff
@@ -906,4 +1007,188 @@ psi::SharedMatrix semicanonicalize_block(psi::SharedWavefunction ref_wfn, psi::S
+                               ref_wfn->options(), scf_type);
+    }
+    JK_actv->set_memory(Process::environment.get_memory() * 0.8);
+
+    JK_actv->initialize();
+    JK_actv->set_do_J(true);
+    JK_actv->set_do_K(true);
+
+    std::vector<std::shared_ptr<psi::Matrix>>& Cl_actv = JK_actv->C_left();
+    std::vector<std::shared_ptr<psi::Matrix>>& Cr_actv = JK_actv->C_right();
+    Cl_actv.clear();
+    Cr_actv.clear();
+    psi::SharedMatrix C_actv_right = psi::linalg::doublet(C_actv, rD_actv, false, false);
+    Cl_actv.push_back(C_actv);       // C^T
+    Cr_actv.push_back(C_actv_right); // rD * C
+
+    JK_actv->compute();
+
+    psi::SharedMatrix F_active(JK_actv->J()[0]->clone());
+    psi::SharedMatrix K_active(JK_actv->K()[0]->clone());
+
+    K_active->scale(0.5);
+    F_active->subtract(K_active);
+    F_active->scale(2.0);
+
+    outfile->Printf("\n Building finished, write AO Fock to wfn ... \n ");
+    psi::SharedMatrix F_tot(F_inactive->clone());
+    F_tot->add(F_active);
+    ref_wfn->Fa()->copy(F_tot);
+}
+
+// Utility function #3: Build the second MO_SPACE_INFO for ASET(2) inner layer computations
+std::shared_ptr<MOSpaceInfo> build_aset2_fragment(psi::SharedWavefunction ref_wfn,
+                                                  std::shared_ptr<MOSpaceInfo> mo_space_info,
+                                                  std::shared_ptr<ForteOptions> options) {
+
+    // int fragment_rvir = options.get_int("FRAGMENT_RUOCC");
+
+    psi::Dimension frzopi = mo_space_info->dimension("FROZEN_DOCC");
+    psi::Dimension nroccpi = mo_space_info->dimension("RESTRICTED_DOCC");
+    psi::Dimension fragment_rocc = mo_space_info->dimension("EMBEDDING_DOCC");
+    psi::Dimension fragment_active = mo_space_info->dimension("EMBEDDING_ACTV");
+    psi::Dimension actv_a = mo_space_info->dimension("ACTIVE");
+    psi::Dimension nrvirpi = mo_space_info->dimension("RESTRICTED_UOCC");
+    psi::Dimension frzvpi = mo_space_info->dimension("FROZEN_UOCC");
+    int add_a_docc = options->get_int("ADD_FRAGMENT_DOCC");
+    int add_a_actv = options->get_int("ADD_FRAGMENT_ACTIVE");
+    bool do_fci = options->get_bool("FRAG_DO_FCI");
+
+    // Write the new active (inner-layer) MOSpaceInfo:
+    std::map<std::string, std::vector<size_t>> mo_space_map_fragment;
+
+    size_t freeze_o = static_cast<size_t>(frzopi[0] + nroccpi[0]);
+    mo_space_map_fragment["FROZEN_DOCC"] = {freeze_o};
+
+    size_t ro = static_cast<size_t>(fragment_rocc[0]);
+    ro += add_a_docc;
+    if (do_fci) {
+        ro = 0;
+    }
+    mo_space_map_fragment["RESTRICTED_DOCC"] = {ro};
+
+    size_t a = static_cast<size_t>(fragment_active[0]);
+    a += add_a_actv;
+    if (do_fci) {
+        a = actv_a[0];
+    }
+    mo_space_map_fragment["ACTIVE"] = {a};
+
+    size_t rv = static_cast<size_t>(
+        actv_a[0] - ro - a); // Read fragment_docc and fragment_active, compute fragment_rvir

```

Comment:
This comment style does not follow the convention in forte.

Reply:
Modified.